### DISCUSSION TITLE
Support `[includeIf "gitdir:/some/path"]`

### DISCUSSION THREAD

**Discussion by little-dude** [magit/magit#4360] [Support] [answered]
Hello,

I have a `~/.config/git/config` so looks contains the following:

```
[user]
        email = "little-dude@mailbox.org"
        name = "little-dude"

[includeIf "gitdir:/home/little-dude/data/work/**"]
        path = "/nix/store/7js3l1sqmlw3z4fr325pd0nc86zx6zan-work.config"
```

The `work.config` file specifies a different `user.email` and `user.name`:

```
❯ cat /nix/store/7js3l1sqmlw3z4fr325pd0nc86zx6zan-work.config
[user]
    name = Corentin Henry
    email = corentin@graphiant.com                        
```

This works well with the `git` CLI:

```
❯ pwd
/home/little-dude/data/work/test
❯ git config --get user.email
corentin@graphiant.com
```

However magit does not seem to support this yet and uses the `user.email` and `user.name` from `~/.config/git/config`.

**Comment by tarsius** [comment, marked as answer]
Magit doesn't have to actively support this; it should just-work™. Basically Magit just runs `git` and does something with the output. The fact that it doesn't just work is due to differences in the environments of the shell and Emacs. If you started Emacs by typing `emacs` in a shell, then you probably wouldn't have this problem.

So in the end this is a configuration issue and I am afraid you'll have to figure it out yourself. (If you learn something that will help other Nix users, then please report it.) Many Emacs users when confronted with variations of this issue choose to not fix the issue and instead paper over it by installing `exec-path-from-shell` and hoping for the best. (Though in this case I suspect that some assembly will be required even when going down that road.)

**Reply by little-dude** [reply]
Thanks a lot for the answer @tarsius. I'm using a combination of [`lorri`](https://github.com/nix-community/lorri) and [`direnv`](https://github.com/direnv/direnv) to set up my shell environment, and [`emacs-direnv`](https://github.com/wbolster/emacs-direnv) to synchronize the shell and emacs. So _a priori_ the emacs environment should be the same, but I'll debug a bit further.

**Reply by little-dude** [reply]
Ok this was trivial to fix actually... So in my case, `~/data/` is a symlink to `/data`. Emacs resolves the link when it opens a file (`pwd` says `/data/work/`), whereas in the shell, `$PWD` is `~/data/work`.

To make it work for both, it's better to specify the resolved path in `includeIf`:
```
[includeIf "gitdir:/data/work/**"]
        path = "/nix/store/7js3l1sqmlw3z4fr325pd0nc86zx6zan-work.config"
```

Sorry for bothering you with a non-magit issue @tarsius! And thank you for magit :)
